src/client.ts

```diff
@@ -435,8 +439,23 @@ export default class Client implements ClientInterface {
         title: "Setting up the bundle",
         cancellable: true,
       },
-      (progress, token) =>
-        this.bundleInstall(customGemfilePath, { progress, token })
+      (progress, token) => {
+        if (this.bundleInstallCancellationSource) {
+          this.bundleInstallCancellationSource.cancel();
```

Comment:
We might want to `.dispose` previous token in this case, otherwise, we might leak memory. So a better call would be:
```suggestion
          this.bundleInstallCancellationSource.dispose(true);
```
Based on the signature of `.dispose` from [here](https://github.com/microsoft/vscode/blob/main/src/vs/base/common/cancellation.ts#L128).

Reply:
Or a sequence of `.cancel()` and `.dispose()` calls, if that API is too new.